### ISSUE TITLE
How many epochs to train

### ISSUE THREAD

**Issue by GimSungwoo** [xiumingzhang/GenRe-ShapeHD#9]
Hi there,

Thanks for sharing the code.

As I don't have a powerful GPU and the dataset is huge, I'm wondering how many epochs roughly does MarrNet2 requires to train? (using all default settings, Adam)

Cheers!

**Comment by xiumingzhang** [comment]
Hi, We didn't follow the traditional definition of an "epoch"; in what I'm about to say below, an "epoch" is not a full pass of all the data, but rather a pass of 10K datapoints.

I took a look at the intermediate visualizations produced by our network training. When we trained MarrNet2 on cars, chairs, and airplanes, 100 epochs gave us reasonable mean shapes of cars, and 200 epochs gave us cars with some coarse details, such as car wheels. For fine details such as car ears, etc., you would need to finetune MarrNet2 with a GAN loss (our ShapeHD paper).

**Comment by GimSungwoo** [comment]
Hi, Xiuming
We trained 300 epoches for MarrNet2 on Chairs and it has converged.
We picked the best.pt and wrote code of IOU to evaluate the MarrNet2 without finetuning, but only get 0.077 IoU on the whole validation set. This is too low. Did you get the similar results for 2.5D to 3D without finetuning? and then get much better IoU after finetuning? if not could you probably share the best marrnet2.pt without finetune.

I'm not sure what goes wrong, I attached the IoU code as below. Did you also set the threshold as 0.5 to binarize voxel values after sigmoid? 

<img width="401" alt="screen shot 2019-03-07 at 12 22 28 pm" src="https://user-images.githubusercontent.com/17474067/53956581-e671b700-40d3-11e9-97e9-2f27e4fa75d0.png">
<img width="575" alt="screen shot 2019-03-07 at 10 45 56 am" src="https://user-images.githubusercontent.com/17474067/53956595-eec9f200-40d3-11e9-93a4-d90dbeaf73fa.png">
